pkg/api/norman/customization/cluster/actions_key.go

```diff
@@ -0,0 +1,71 @@
+package cluster
+
+import (
+	"bytes"
+	"encoding/json"
+	"net/http"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/rancher/norman/api/access"
+	"github.com/rancher/norman/types"
+	v3 "github.com/rancher/rancher/pkg/apis/management.cattle.io/v3"
+	v3client "github.com/rancher/rancher/pkg/client/generated/management/v3"
+	"github.com/rancher/rancher/pkg/controllers/management/etcdbackup"
+	mgmtv3 "github.com/rancher/rancher/pkg/generated/norman/management.cattle.io/v3"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func (a ActionHandler) RotateEncryptionKey(actionName string, action *types.Action, apiContext *types.APIContext) error {
+	response := map[string]interface{}{
+		"type": v3client.RotateEncryptionKeyOutputType,
+		v3client.RotateEncryptionKeyOutputFieldMessage: "etcd backup created, rotating encryption key",
+	}
+
+	var mgmtCluster mgmtv3.Cluster
+	if err := access.ByID(apiContext, apiContext.Version, apiContext.Type, apiContext.ID, &mgmtCluster); err != nil {
+		response[v3client.RotateEncryptionKeyOutputFieldMessage] = "cluster does not exist"
+		apiContext.WriteResponse(http.StatusBadRequest, response)
+		return errors.Wrapf(err, "failed to get cluster by ID %s", apiContext.ID)
+	}
+
+	cluster, err := a.ClusterClient.Get(apiContext.ID, v1.GetOptions{})
+	if err != nil {
+		response[v3client.RotateEncryptionKeyOutputFieldMessage] = "cluster does not exist"
+		apiContext.WriteResponse(http.StatusBadRequest, response)
+		return errors.Wrapf(err, "failed to get cluster by ID %s", apiContext.ID)
+	}
+
+	// create etcd backup so that if the rotate encryption action fails, the user can restore from this snapshot
```

Comment:
qq: setting RotateEncryptionKey field true immediately triggers RKE, do we need to wait for backup to complete for failure usecases? or rotate can work in parallel?

Reply:
I've removed the etcd backup logic from the key rotation action, we will handle ensuring the user has created a backup via a UI modal after the action is clicked.